domains/Links/Observers/LinkObserver.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+
+namespace Domains\Links\Observers;
+
+
+use Domains\Links\Exceptions\UnapprovedLinkLimitException;
+use Domains\Links\Models\Link;
+
+class LinkObserver
+{
+    public function creating(Link $link)
+    {
+        $pendingCount = Link::forAuthorWithEmail($link->author_email)
```

Comment:
Good work, here.

I just think we can avoid the static call of the Link model, here, since we can infer and create a new instance of that model based on the injected `$link` variable, this way: `$link->newModelQuery()`.

Reply:
Ok, but for my knowledge, what is the advantage?

Reply:
Just re-using the instance we already have. It avoids having to go through all the model booting process.